src/Drupal/Commands/core/WatchdogCommands.php

```diff
@@ -62,9 +64,30 @@ public function show($substring = '', $options = ['format' => 'table', 'count' =
             $row = $this->formatResult($result, $options['extended']);
             $table[$row->wid] = (array)$row;
```

Comment:
This initial set of records just hangs out in memory until you exit the tail process. I think you should check for $tail earlier to avoid selecting them initially.